README.md

```diff
@@ -37,85 +39,115 @@ This library supports both **Swift** and **Objective-C**.
 
 #### CocoaPods
 
-Sentinel is available through [CocoaPods](https://cocoapods.org). To install
+*Sentinel* is available through [CocoaPods](https://cocoapods.org). To install
 it, simply add the following line to your Podfile:
 
 ```ruby
 pod 'Sentinel'
 ```
 
+*Sentinel* is made of multiple subspecs:
+- `Core` which will install only the core features for the *Sentinel* to be usable
+- `UserDefaults` which will add the `UserDefaultsTool`
+- `EmailSendel` which will add the `EmailSenderTool`
+- `CustomLocation` which will add the `CustomLocationTool`
+- `TextEditing` which will add the `TextEditingTool`
+- `Default` - which will install `UserDefaults`, `Core`, `CustomLocation`, and `TextEditing`
+
+*NOTE: All of the subspecs add `Core` as a dependency*
+
 #### Swift Package Manager
 
 If you are using SPM for your dependency manager, add this to the dependencies in your `Package.swift` file:
 
 ```swift
 dependencies: [
-    .package(url: "https://github.com/infinum/ios-sentinel.git")
+.package(url: "https://github.com/infinum/ios-sentinel.git")
 ]
 ```
 
-To add the *Sentinel* to your project, install the library as instructed above. After that, define *Sentinel* configuration with mandatory parameters and call `Sentinel.shared.setup:` in the `AppDelegate` method `application(_:didFinishLaunchingWithOptions:)`.
+## Usage
+
+`Sentinel` is the main class used to set up the *Sentinel* which will be used in the application. The `Sentinel` object can configured via `setup:` method by `Configuration` object. The `setup:` can be called in the `AppDelegate` method `application(_:didFinishLaunchingWithOptions:)`.
 
 ```swift
-let configuration = Sentinel.Configuration(
-    sourceScreenProvider: SourceScreenProviders.default,
-    trigger: Triggers.shake,
-    tools: [
-        GeneralInfoTool(),
-        UserDefaultsTool(),
-        LoggieTool(),
-    ],
-    preferences: [
-        OptionSwitchItem(
-            name: "Analytics",
-            setter: { AppSwitches.analyticsEnabled = $0 },
-            getter: { AppSwitches.analyticsEnabled },
-            userDefaults: .standard,
-            userDefaultsKey: "com.infinum.sentinel.optionSwitch.analytics"
-        )
-    ]
-)
-
-Sentinel.shared.setup(with: configuration)
+    let configuration = Sentinel.Configuration(
+        trigger: Triggers.shake,
+        tools: [
+            UserDefaultsTool(),
+            CustomLocationTool()
+        ],
+        preferences: optionSwitchItems
```

Comment:
Please add `optionSwitchItems` somewhere because it is hard to understand what they are in this example.